File: scripts/templates/python/part2.py
import sys
import time

def read_input(file_path):
    with open(file_path, "r") as f:
        return f.read().splitlines()

def solve(input_data):
    # Your solution for part 2 goes here
    return "solution for part 2"

if __name__ == "__main__":
    start_time = time.time()
    input_file = "input.txt"
    if len(sys.argv) > 1:
        input_file = sys.argv[1]
    
    input_data = read_input(input_file)
    solution = solve(input_data)
    
    end_time = time.time()
    
    print(f"Solution: {solution}")
    print(f"Completed in {end_time - start_time:.4f} seconds")

File: coders/erik/day_01/src/part2.rs
use std::env;
use std::fs::File;
use std::io::{self, BufRead};
use std::path::Path;
use std::time::Instant;

fn read_lines<P>(filename: P) -> io::Result<Vec<String>>
where
    P: AsRef<Path>,
{
    let file = File::open(filename)?;
    io::BufReader::new(file).lines().collect()
}

fn solve(input: &[String]) -> String {
    let mut dial = 50;
    let mut count = 0;

    for line in input {
        let direction = &line[0..1];

        let mut num : i32 = line[1..].parse().unwrap();

        let numOfFullRotations = ((num as f64)/100.0).floor().abs() as i32;
        count += numOfFullRotations;

        num = num % 100;

        if ( direction == "L") {
            dial = dial - num;
            if dial < 0 && num != dial.abs() { count += 1 }
        }
        else {
            dial = dial + num;
            if dial > 100 && num != 100 - dial { count += 1 }
        }

        dial = dial % 100;


        if dial == 0 {
            count += 1;
        }

        if dial < 0 {
            dial = 100 + dial
        }

    }

    return count.to_string();
}

fn main() {
    let start_time = Instant::now();

    let args: Vec<String> = env::args().collect();
    let input_file = if args.len() > 1 { &args[1] } else { "input.txt" };

    if let Ok(lines) = read_lines(input_file) {
        let solution = solve(&lines);
        let duration = start_time.elapsed();
        println!("Solution: {}", solution);
        println!("Completed in {:.4} seconds", duration.as_secs_f64());
    } else {
        eprintln!("Error reading input file");
    }
}
